### PR TITLE
Move ApplicationResource endpoint #681

### PR DESCRIPTION
**Pull request by pmi**

### CODE REVIEW

File: src/main/java/com/enonic/xp/app/users/rest/AdminCmsRestActivator.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @Deactivate
+    public void deactivate()
+    {
+        this.service.destroy();
+    }
+
+    @Reference
+    public void setService( final JaxRsServiceFactory factory )
+    {
+        this.service = factory.newService( "v2", "/admin/v2/rest", DispatchConstants.XP_CONNECTOR );
```
Comment (rymsha) [inline src/main/java/com/enonic/xp/app/users/rest/AdminCmsRestActivator.java:32]: I think, it is a collision here. If all apps try to register JaxRsService "v2" service, on the same path, only one service would work.
So, paths service paths, and groups should be different

File: src/main/java/com/enonic/xp/app/users/rest/resource/application/ApplicationIconUrlResolver.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import com.enonic.xp.app.ApplicationDescriptor;
+import com.enonic.xp.app.ApplicationKey;
+import com.enonic.xp.app.users.rest.resource.schema.IconUrlResolver;
+import com.enonic.xp.icon.Icon;
+
+
+public final class ApplicationIconUrlResolver
+    extends IconUrlResolver
+{
+    private static final String REST_SCHEMA_ICON_URL = "/admin/rest/application/icon/";
```
Comment (rymsha) [inline src/main/java/com/enonic/xp/app/users/rest/resource/application/ApplicationIconUrlResolver.java:12]: It seems to be an old path

**Review by rymsha — CHANGES_REQUESTED** [review]

**Review by rymsha — APPROVED** [review]

File: src/main/java/com/enonic/xp/app/users/rest/AdminCmsRestActivator.java (hunk 1 above)
Comment (pmi) [reply src/main/java/com/enonic/xp/app/users/rest/AdminCmsRestActivator.java]: Strange thing, it didn't work at all when I changed groups and paths to be different in all apps. 
What I did was:
In AdminCmsRestActivator.java
`this.service = factory.newService( "v2users", ResourceConstants.REST_ROOT, DispatchConstants.XP_CONNECTOR );`
where
`String REST_ROOT = "admin/v2/rest/users/";`
And then in resource:
`@Component(immediate = true, property = "group=v2users")`

File: src/main/java/com/enonic/xp/app/users/rest/resource/ResourceConstants.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package com.enonic.xp.app.users.rest.resource;
+
+public interface ResourceConstants
+{
+    String REST_ROOT = "admin/v2/rest/users/";
```
Comment (rymsha) [inline src/main/java/com/enonic/xp/app/users/rest/resource/ResourceConstants.java:5]:
```suggestion
    String REST_ROOT = "/admin/rest-v2/users/";
```

**Review by rymsha — CHANGES_REQUESTED** [review]

**Review by rymsha — APPROVED** [review]

**Review by alansemenov — APPROVED** [review]

**Pull request merged**